targets/curie_bsp/setup.py

```diff
@@ -159,62 +170,83 @@ def create_makefile_in_curie(curie_path):
 VERSION_MINOR  := 0
 VERSION_PATCH  := 0
 include $(T)/build/project.mk
-'''
-    file_path = os.path.join(curie_path, 'wearable_device_sw/projects/' \
-                                        + project_name + '/Makefile')
-    file_to_be_created = open(file_path, 'w+')
-    file_to_be_created.write(out_put)
-    file_to_be_created.close()
-
-    # Kbuild.mk in arc/
-    breadth_first_travel(os.path.join(curie_path, 'wearable_device_sw/projects/' \
-                                                + project_name + '/arc/'))
-    # Kbuild.mk in quark/
-    file_path = os.path.join(curie_path, 'wearable_device_sw/projects/' \
-                                        + project_name + '/quark/Kbuild.mk')
-    out_put = '''
-obj-y += jerryscript/
+'''.format(project_name=project_name)
+
+    write_file(root_makefile_path, root_makefile_content)
+
+
+def create_arc_kbuild(project_path):
+    """ Creates @project_path/arc/Kbuild.mk file. """
+
+    arc_path = os.path.join(project_path, 'arc')
+    arc_kbuild_path = os.path.join(arc_path, 'Kbuild.mk')
+    arc_sources = find_sources(arc_path, '.')
+    arc_kbuild_content = build_obj_y(arc_sources)
+
+    write_file(arc_kbuild_path, arc_kbuild_content)
+
+
+def create_quark_kbuild(project_path, jerry_path):
+    """ Creates @project_path/quark/Kbuild.mk file. """
+    quark_kbuild_path = os.path.join(project_path, 'quark', 'Kbuild.mk')
+
+    # Extract a few JerryScript related data
+    jerry_data = build_jerry_data(jerry_path)
+    jerry_objects = build_obj_y(jerry_data['sources'])
+    jerry_defines = jerry_data['cflags']
+    jerry_build_dirs = build_mkdir(jerry_data['dirs'])
+
+    quark_include_paths = [
+        'include',
+        'jerryscript',
+        'jerryscript/jerry-libm/include',
+        'jerryscript/targets/curie_bsp/include'
+    ] + list(jerry_data['dirs'])
+
+    quark_includes = [
+        '-Wno-error',
+    ] + ['-I$(PROJECT_PATH)/quark/%s' % dir for dir in quark_include_paths]
```

Comment:
I wanted to keep the generated code as close as possible to the previous version. Also I don't know if there would be any side effect if we change it to the `project_path` one.